src/Features/Core/Portable/AddMissingReference/AddMissingReferenceCodeAction.cs

```diff
@@ -53,11 +52,12 @@ public static async Task<CodeAction> CreateAsync(Project project, AssemblyIdenti
             // whatever project reference we end up adding won't add a circularity (also good.)
             foreach (var candidateProjectId in dependencyGraph.GetProjectsThatThisProjectTransitivelyDependsOn(project.Id))
             {
-                var candidateProject = project.Solution.GetProject(candidateProjectId);
-                if (string.Equals(missingAssemblyIdentity.Name, candidateProject.AssemblyName, StringComparison.OrdinalIgnoreCase))
+                var candidateProject = project.Solution.GetRequiredProject(candidateProjectId);
+                if (candidateProject.SupportsCompilation &&
+                    string.Equals(missingAssemblyIdentity.Name, candidateProject.AssemblyName, StringComparison.OrdinalIgnoreCase))
```

Comment:
@jasonmalinowski to verify if we need to add any new checks for target projects that do not support compilation